framework/src/modules/chain/processor/processor.js

```diff
@@ -293,34 +293,41 @@ class Processor {
 				skipExistingCheck: skipSave,
 				tx,
 			});
+
 			if (!skipBroadcast) {
 				this.channel.publish('chain:processor:broadcast', {
 					block: cloneDeep(block),
 				});
 			}
+
+			if (!skipSave) {
+				const blockJSON = await this.serialize(block);
+				await this.blocksModule.save({ blockJSON, tx });
+			}
+
+			// Apply should always be executed after save as it performs database calculations
```

Comment:
```suggestion
			// Apply should always be executed after save as it performs database calculations
			// because sumround in the dpos expects to have this processing block in the database in apply step
```